### DISCUSSION TITLE
[console]Lockable trait locking commands with the same name across instances on the same machine

### DISCUSSION THREAD

**Discussion by redheness** [symfony/symfony#52837] [Q&A]
### Symfony version(s) affected

6.* and 7.*

### Description

If you host two instances of a project on the same machine and one of it's command use the [LockableTrait](https://github.com/symfony/console/blob/6.3/Command/LockableTrait.php) to prevent two execution of the same command. It will lock the command on both instances.

### How to reproduce

- Create a project with a command and use the lockable trait
- Duplicate the project on the same machine
- Try to execute the same command at the same time on each instances

### Possible Solution

In [Symfony\Component\Console\Command\LockableTrait](https://github.com/symfony/console/blob/6.3/Command/LockableTrait.php)

Replace
```PHP
$this->lock = (new LockFactory($store))->createLock($name ?: $this->getName());
```

with
```PHP
$this->lock = (new LockFactory($store))->createLock($name ?: getcwd() . `|` . $this->getName());
```

**Comment by nicolas-grekas** [comment]
I'd say this is still the expected behavior. The trait is just a helper for common needs but if yours are more specific, you may override the lock() method with whatever logic you'd like. E.g. the one you're suggesting, or using a completely different lock factory for example.
Not using the trait and implementing everything on your side is also an option of course.
